File: sol/solution/0201-0300/0201/Solution.php
class Solution {
    /**
     * @param Integer $left
     * @param Integer $right
     * @return Integer
     */
     function rangeBitwiseAnd($left, $right) {
        while ($left < $right) {
            $right &= ($right - 1);
        }
        return $right;
    }
}

File: sol/solution/0201-0300/0201/Solution.ex
defmodule Solution do
  use Bitwise
  @spec range_bitwise_and(left :: integer, right :: integer) :: integer
  def range_bitwise_and(left, right) when left < right do
    range_bitwise_and(left, right &&& (right - 1))
  end
  def range_bitwise_and(_, right), do: right
end